deepchecks/tabular/checks/data_integrity/feature_feature_correlation.py

```diff
@@ -0,0 +1,102 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""module contains Correlated Features check."""
+
+import pandas as pd
+import plotly.express as px
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils.correlation_methods import correlation_ratio, symmetric_theil_u_correlation
+from deepchecks.utils.dataframes import generalized_corrwith
+
+__all__ = ['FeatureFeatureCorrelation']
+
+
+class FeatureFeatureCorrelation(SingleDatasetCheck):
+    """Checks for pairwise correlation between the features.
+
+        Extremely correlated pairs could indicate redundancy and even duplication.
+
+    Parameters
+    ----------
+        n_top_pairs : int , default: 5
+        Number of pairs to show, sorted by the correlation strength
+    """
+
+    def __init__(self,
+                 n_top_pairs: int = 5,
+                 **kwargs):
+        super().__init__()
+        self.n_top_pairs = n_top_pairs
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """
+        Run Check.
+
+        Returns
+        -------
+        CheckResult
+        A heatmap of the pairwise correlations between the features
+        """
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+
+        dataset.assert_features()
+        num_features = dataset.numerical_features
+        cat_features = dataset.cat_features
+
+        encoded_cat_data = dataset.data.loc[:, cat_features].apply(lambda x: pd.factorize(x)[0])
+
+        # Numerical-numerical correlations
+        num_num_corr = dataset.data.loc[:, num_features].corr(method='spearman')
+
+        # Numerical-categorical correlations
+        num_cat_corr = generalized_corrwith(dataset.data.loc[:, num_features], encoded_cat_data,
+                                            method=correlation_ratio)
```

Comment:
why use this generalized version when .corr gets Callable as method as well? What's the difference?

Reply:
.corr is with self, .corrwith is with another dataframe with the same column names, generalized_corrwith is between any 2 dataframes with the same number of rows

Reply:
Ohhh ok, thanks :)